File: app/src/main/java/com/railian/maksym/provectustesttask/Location.java

package com.railian.maksym.provectustesttask;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Location {

    // Applying annotations for gson convertor
    @SerializedName("street")
    @Expose
    private String street;
    @SerializedName("city")
    @Expose
    private String city;
    @SerializedName("state")
    @Expose
    private String state;
    @SerializedName("postcode")
    @Expose
    private String postcode;

    // Constructor for using in constructor
    public Location(String street,String city,String state, String postcode){
        this.street=street;
        this.city=city;
        this.state=state;
        this.postcode=postcode;

    }

    // Getters and Setters
    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getPostcode() {
        return postcode;
    }

    public void setPostcode(String postcode) {
        this.postcode = postcode;
    }

}

File: app/src/main/java/com/railian/maksym/provectustesttask/models/Location.kt
package com.railian.maksym.provectustesttask.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Location(@field:SerializedName("street")
               @field:Expose
               val street: String, @field:SerializedName("city")
               @field:Expose
               val city: String, @field:SerializedName("state")
               @field:Expose
               val state: String, @field:SerializedName("postcode")
               @field:Expose
               val postcode: String)

